src/index.ts

```diff
@@ -407,27 +410,42 @@ async function run() {
 
     // Try to figure out whoami
     await getWhoami();
+
+    // This covers the case where the meeting is in progress and this script was restarted. 
+    // Let everyone know in the mmeting that the Primary was restarted and they should send /whoami
+    // so oscdevices can be restored.  
+    // Everyone needs to be asked, because the Primary no longer knows who is running ZoomOSC
+    if (primaryMode && initialCallStatus) {
+        sendToZoom('/zoom/all/chat', `/primaryrestarted`);
+    }
 }
 
 async function getWhoami() {
+
+    await sleep(2000);
+
     if (primaryMode) {
         // /list command will update myZoomID
         sendToZoom('/zoom/list');
         await sleep(2000);
     } else {
         // Ask the Primary user whoami
+        // If the Primary is online, the /list response will update the myZoomID fairly quickly, If not, poll until it is.
+        // There is nothing else to do until whoami is known
         sendToZoom('/zoom/userName/chat', PRIMARY_USER, `/whoami`);
-
         await sleep(2000);
-        if (myName) {
-            sendToZoom('/zoom/list');
+        while (!myZoomID) {
```

Comment:
this does the job. if you ever wanted to try an async way of doing it, it might get wrapped in something using setTimeout(). It might need more restructuring to do that though. I wish I had more time to play with this. I like working out how to make things work asyncrhonously.